File: src/com/beyondbell/connectFour/Piece.java
package com.beyondbell.connectFour;

import org.jetbrains.annotations.Contract;

import java.awt.Color;


public enum Piece {
	Empty(Color.WHITE), Yellow(Color.YELLOW), Red(Color.RED);

	private final Color color;

	Piece(final Color color) {
		this.color = color;
	}

	@Contract(pure = true)
	public final Color getColor() {
		return color;
	}
}

File: src/main/java/com/beyondbell/connectFourAPI/Piece.kt
package com.beyondbell.connectFourAPI

enum class Piece {
	Empty, Yellow, Red;

	fun getWinner(): WinState {
		return when (this) {
			Yellow -> WinState.Yellow
			Red -> WinState.Red
			else -> WinState.NoOne
		}
	}
}